subsys/net/ip/net_if.c

```diff
@@ -3990,7 +3990,11 @@ int net_if_up(struct net_if *iface)
 	/* In many places it's assumed that link address was set with
 	 * net_if_set_link_addr(). Better check that now.
 	 */
-	NET_ASSERT(net_if_get_link_addr(iface)->addr != NULL);
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_CAN)) {
```

Comment:
I don't think that's enough, with such condition we'll skip the LL address verification for any interface, not only the CAN ones.

@alexanderwachter In the corresponding issue, you've mentioned `is_socket_can(iface)`, I don't see such a function in the tree though? Was this just a suggestion to create one or does the function exist somewhere under a different name? I think such a functionality could be achieved with the following code snippet as well?:
```
net_if_l2(iface) == &NET_L2_GET_NAME(CANBUS)
```

Reply:
There is no such function yet. That was just an example as pseudo code